src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -340,7 +365,78 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
         }
     }
 
-    private long writePQ(SequentialWriter writer) throws IOException
+    private boolean isFastFindRows(BiMap<Integer, Integer> ordinalMap)
+    {
+        boolean canFastFindRows;
+        canFastFindRows = true;
+        int minRow = Integer.MAX_VALUE;
+        int maxRow = Integer.MIN_VALUE;
+        for (VectorPostings<T> vectorPostings : postingsMap.values())
+        {
+            if (vectorPostings.getRowIds().size() != 1)
+            {
+                canFastFindRows = false;
+                ordinalMap.clear();
+                break;
+            }
+            int rowId = vectorPostings.getRowIds().getInt(0);
+            int ordinal = vectorPostings.getOrdinal();
+            minRow = Math.min(minRow, rowId);
+            maxRow = Math.max(maxRow, rowId);
+            if (ordinalMap.containsKey(ordinal))
+            {
+                canFastFindRows = false;
+                ordinalMap.clear();
+                break;
+            } else {
+                ordinalMap.put(ordinal, rowId);
+            }
+        }
+
+        if (minRow != 0 || maxRow != postingsMap.values().size() - 1)
+        {
+            canFastFindRows = false;
+            ordinalMap.clear();
+        }
+        return canFastFindRows;
+    }
+
+    // taken and adopted from com.github.jbellis.jvector.disk.OnDiskGraphIndex
+    private static <T> void writeGraph(GraphIndex<T> graph,
```

Comment:
As noted earlier, I'd prefer to implement a GraphIndex (in the C* codebase, not jvector) that does the remapping

Reply:
I'd do this as a separate change as it requires change in jvector/release/update to that release/change in c*.
First we need to agree on this approach/functionality

Reply:
What changes do you need in jvector?

Reply:
I wanted to implement this as a write method in jvector to avoid extra levels of abstractions / creation of new objects.
Anyways, here it is.